lib/crypto/crypto_keys_subr.c

```diff
@@ -32,7 +32,7 @@
  */
 
 static int import_BN(BIGNUM **, const uint8_t **, size_t *);
-static int export_BN(BIGNUM *, uint8_t **, size_t *, uint32_t *);
+static int export_BN(const BIGNUM *, uint8_t **, size_t *, uint32_t *);
```

Comment:
This is problematic because old versions of OpenSSL declare BN_bn2bin as taking a `BIGNUM *` argument instead of a `const BIGNUM *` argument.  Admittedly, it looks like "old versions" in this case means pre-0.9.4, but I wouldn't be surprised if other crypto libraries copied the broken API.

How about deconsting bn in the call to BN_bn2bin?